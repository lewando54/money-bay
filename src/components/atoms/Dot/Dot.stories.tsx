/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import Dot from './Dot'

const meta = {
	component: Dot,
	argTypes: {
        // To write
	}
} satisfies Meta<typeof Dot>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
        // To write
	}
}
