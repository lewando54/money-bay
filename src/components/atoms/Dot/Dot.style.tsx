import { StyleSheet } from 'react-native'
import { MAIN_DARK } from '../../../styling/GlobalTheme.style'
export default StyleSheet.create({
    dot: {
        width: 8,
        height: 8,
        borderRadius: 100,
        borderWidth: 3
    }
})
