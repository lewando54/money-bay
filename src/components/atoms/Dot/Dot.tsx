import React from 'react'
import dotStyle from './Dot.style'
import Animated, { Extrapolate, SharedValue, interpolate, useAnimatedStyle } from 'react-native-reanimated'
import { Dimensions } from 'react-native'
import { MAIN_DARK } from '../../../styling/GlobalTheme.style'

interface IDotProps {
    index?: number
    x?: SharedValue<number>
    color?: string
    testID?: string
}

export default function Dot({
    index,
    x,
    color=MAIN_DARK,
    testID
}: IDotProps){
    const { width } = Dimensions.get('window')

    const animatedDotStyle = x && useAnimatedStyle(() => ({
        opacity: interpolate(
            x.value, 
            [
                (index - 1) * width,
                index * width,
                (index + 1) * width
            ], 
            [0.2, 1, 0.2],
            Extrapolate.CLAMP
        )
    }))

    return (
        <Animated.View style={[dotStyle.dot, {borderColor: color}, x && animatedDotStyle]} />
    )
}
