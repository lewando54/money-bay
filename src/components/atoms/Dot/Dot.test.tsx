import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import Dot from './Dot'
describe('Dot', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Dot />)
    expect(tree).toMatchSnapshot()
  })
})
