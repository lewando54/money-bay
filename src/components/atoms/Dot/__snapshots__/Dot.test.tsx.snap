// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot should render correctly 1`] = `
<View
  animatedStyle={
    {
      "value": {},
    }
  }
  collapsable={false}
  style={
    {
      "borderColor": "#1B1D4D",
      "borderRadius": 100,
      "borderWidth": 3,
      "height": 8,
      "width": 8,
    }
  }
/>
`;
